src/installer/managed/Microsoft.NET.HostModel/MachO/BinaryFormat/Blobs/SimpleBlob.cs

```diff
@@ -0,0 +1,49 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Diagnostics;
+using System.IO;
+using System.IO.MemoryMappedFiles;
+using Microsoft.NET.HostModel.MachO;
+
+/// <summary>
+/// This class represents a simple blob with a byte array data.
+/// </summary>
+internal class SimpleBlob : Blob
+{
+    public override uint Size => base.Size + (uint)_data.Length;
+    protected byte[] _data;
+
+    public SimpleBlob(MemoryMappedViewAccessor accessor, long offset) : base(accessor, offset)
+    {
+        accessor.Read(offset + sizeof(uint), out uint size);
```

Comment:
Doing I/O in a constructor is generally bad practice. Constructors can't return errors or be async, so you heavily narrow your options. Most users will also not expect constructors to do significant work.